pkg/query-service/app/querier/helper.go

```diff
@@ -80,6 +81,78 @@ func prepareLogsQuery(_ context.Context,
 	return query, err
 }
 
+// New function to filter series points
+func filterSeriesPoints(seriesList []*v3.Series, missStart, missEnd int64, stepInterval int64) ([]*v3.Series, int64, int64) {
```

Comment:
This should be shared instead of copying here because I don't see any difference?

Reply:
yeah, better to move it common